src/libprojectM/Renderer/hlslparser/src/HLSLTree.h

```diff
@@ -513,6 +517,22 @@ struct HLSLArgument : public HLSLNode
     bool                    hidden;
 };
 
+/** Macro declaration */
+struct HLSLMacro : public HLSLStatement
+{
+    static const HLSLNodeType s_type = HLSLNodeType_Macro;
+    HLSLMacro()
+    {
+        name            = NULL;
+        argument        = NULL;
+        macroAliased    = NULL;
+    }
+    const char*         name;
+    HLSLArgument*       argument;
+    std::string         value;
```

Comment:
this errors for me

<img width="790" alt="screen shot 2018-07-23 at 13 20 35" src="https://user-images.githubusercontent.com/245131/43071381-53ababf2-8e7b-11e8-961c-051783f29b60.png">

Reply:
easily fixed by #including  `<string>`